src/realm/object-store/sync/app_utils.cpp

```diff
@@ -22,18 +22,38 @@
 
 #include <external/json/json.hpp>
 
+#include <algorithm>
+
 namespace realm {
 namespace app {
 
+static bool string_iequals(const std::string& a, const std::string& b)
+{
+    return std::equal(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+        return tolower(a) == tolower(b);
+    });
+}
+
+util::Optional<std::pair<const std::string, std::string>>
```

Comment:
```suggestion
std::pair<const std::string, std::string>*
```

No reason to copy the result. Calling code shouln''t need to be modified.